### PR TITLE
[release-v3.27] Auto pick #8388: Add tests for DefaultEndpointToHostAction.

### PR DESCRIPTION
**Pull request by tomastigera**
Cherry pick of #8388 on release-v3.27.

#8388: Add tests for DefaultEndpointToHostAction.

# Original PR Body below

## Description

If there is no wildcard HEP, there is no policy that should be applied. But without skipping, empty list of profiles would create a default deny rule if none of the non-existent profiles matches. That is obviously always hit and traffic toward the host is dropped if defaultEndpointToHostAction is set to RETURN.

In fact, to be compatible with iptables behaviour, we should never apply normal *-hep policy, only the preDNAT one. This PR leaves the normal policy code there for enabling it in the future.

Also includes FV tests for defaultEndpointToHostAction by @fasaxc that codify behaviour and unify it across dataplanes.

Removed blanket ACCEPT in INPUT chain, only wg traffic is explicitly accepted now, the rest is dropped if INPUT policy is DROP and defaultEndpointToHostAction==RETURN.
## Related issues/PRs

fixes https://github.com/projectcalico/calico/issues/7252
## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
ebpf: align defaultEndpointToHostAction with iptables - do not apply normal *-hep policy to wep
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Review by sridhartigera — APPROVED** [review]

**Pull request merged**
